script.js

```diff
@@ -212,13 +247,78 @@ function setBoard(){
   $('.log').html('');
   selects = [];
   makeDeck();
-  cards = shuffleArray(deck);
+  shuffleAndFindNCards(12);
   for(var i=0;i<12;i++){
     $('.board').append(drawCard(getNextCard()));
   }
   animateCards(100);
 }
 
+// Shuffles the deck until the first N cards, if dealt onto the existing
+// board (minus cardsToRemove, if provided), would create disjoint SET(s) and
+// only disjoint SET(s).
+function shuffleAndFindNCards(n, cardsToRemove) {
+  cardsToRemove = cardsToRemove || [];
+  let board = Array.from($('.board .card')).map(card => $(card).data());
+  board = board.filter(card => !cardsToRemove.includes(card));
+  const iterationLimit = 1000;
+  for (var i = 0; i < iterationLimit; i++) {
+    shuffleArray(deck);
+    const firstNCards = deck.slice(0, n);
+    if (hasDisjointMatchesOnly(board.concat(firstNCards))) {
+      console.debug(`${i} shuffles`);
+      return;
+    }
+  }
+  console.debug(`Failed after ${iterationLimit} shuffles`);
+}
+
+function findAllMatches(board) {
+  const result = [];
+  for (let cards of combinations(board, three)) {
+    if (checkIfSET(cards)) {
+      result.push(cards);
+    }
+  }
+  return result;
+}
+
+function findAllDisjointMatches(board) {
+  const result = [];
+  for (let cards of combinations(board, three)) {
+    const { isSET, differences } = checkIfSET(cards, 'countDifferences')
+    if (isSET && differences === 4) {
```

Comment:
At some point, I'm sure I'll want to make a hard mode that allows SETs with 3 or 4 differences, or a totally customizable game. Here's how:

```suggestion
    if (isSET && differences >= 3) {
```